sublist/Cargo.toml

```diff
@@ -7,7 +7,8 @@ version = "0.0.0"
 rayon = "1.6.1"
 
 [dev-dependencies]
-criterion = { version = "0.3", features = ["html_reports"]}
+criterion = { version = "0.4", features = ["html_reports"]}
```

Comment:
This is kind of unrelated, but maybe still interesting:

This was the first time I had trouble with incompatible dependency versions. The direct dependency on criterion was pulling in `3.6`, but pprof required `4.0`.

I then got an error on the line where I was passing the pprof profiler to the criterion config:

```rust
Criterion::default().with_profiler(PProfProfiler::new(100, Output::Flamegraph(None)))
```

Here's the error message:

> the trait bound `pprof::criterion::PProfProfiler<'_, '_>: criterion::profiler::Profiler` is not satisfied
perhaps two different versions of crate `criterion` are being used?
the trait `criterion::profiler::Profiler` is implemented for `criterion::profiler::ExternalProfiler`

This error message is mostly cryptic, but it does include a note that incompatible versions might be the problem. I then text-searched the `Cargo.lock` for occurrances of "criterion" and found two versions, `3.6` and `4.0`.

I thought I include this little debugging timeline in case it's interesting to you :-)

Reply:
good to know, thanks